hack/parse-prow-creds.sh

```diff
@@ -34,3 +34,11 @@ if [[ -n "${AZURE_CREDENTIALS:-}" ]]; then
 
     export AZURE_SUBSCRIPTION_ID AZURE_TENANT_ID AZURE_CLIENT_ID AZURE_CLIENT_SECRET AZURE_STORAGE_ACCOUNT AZURE_STORAGE_KEY
 fi
+
+# for prow we use the provided AZURE_CREDENTIALS_WORKLOAD file.
+# the file is expected to be in toml format.
+if [[ -n "${AZURE_CREDENTIALS_WORKLOAD_ID:-}" ]]; then
+    AZURE_WORKLOAD_ID="$(parse_cred ClientID < "${AZURE_CREDENTIALS_WORKLOAD_ID}")"
+    # TODO add federated identity file location
```

Comment:
Wondering if we should be saving the `federated token as well` in the prow config? @jsturtevant @mboersma @jackfrancis  ?

Reply:
Do we want to have a separate mounted secret for this new workload-identity input @jsturtevant? Alternatively, we could:

- pass in the workload id directly as a prow env var
- add to the existing mounted secret located at the filepath `$AZURE_CREDENTIALS` (L27 above)

Reply:
>Wondering if we should be saving the federated token as well in the prow config? @jsturtevant @mboersma @jackfrancis ?

No, we need to read it from that file each time.  The Federated token is generated by the Kubernetes API server for each Pod and it is destroyed when the pod is removed.  Think of it as an on-demand secret provided when the pod is created. 

> Do we want to have a separate mounted secret for this new workload-identity input @jsturtevant? Alternatively, we could:

I think adding the file pointing to a secret in the PROW environment will be a lot of overhead for values that are not secret and it would make easier to maintain by just using prow envs

Reply:
@nawazkh following James's recommendation, looks like we can remove all changes to this file. We can get `$AZURE_WORKLOAD_ID` directly from the prow job's environment.

Reply:
> Think of it as an on-demand secret provided when the pod is created. 

So do we need to have two different paths hardcoded and saved somewhere, say prow env, and consume it based on the type of community cluster?
Or maybe its a future story to address when we move the tests to community clusters?

Reply:
The path will be the same, because it is in the Prow configuration.  The prow configuration gets translated to a Pod configuration with a volume mount.  See https://github.com/kubernetes/test-infra/pull/32696  as an example.

The path will always be `/var/run/secrets/azure-token/serviceaccount/token` inside the pod

Reply:
Created a test infra draft PR with Azure Workload ID as env variable, please take a look https://github.com/kubernetes/test-infra/pull/32837